mlir/test/Dialect/SCF/loop-pipelining.mlir

```diff
@@ -802,8 +801,10 @@ func.func @stage_0_value_escape(%A: memref<?xf32>, %result: memref<?xf32>, %ub:
 //        CHECK:   } else {
 //        CHECK:   }
 //        CHECK:   return
-func.func @dynamic_loop(%A: memref<?xf32>, %result: memref<?xf32>, %lb: index, %ub: index, %step: index) {
+func.func @dynamic_loop(%A: memref<?xf32>, %result: memref<?xf32>, %ub: index) {
   %cf = arith.constant 1.0 : f32
+  %lb = arith.constant 3 : index
+  %step = arith.constant 2 : index
```

Comment:
why do we want those to be constants?

Reply:
Just added to make a lower_bound non-zero. The loop is still dynamic with the upper_bound passed in.

Reply:
wasn't it already a lower_bound non-zero?

Reply:
It helps demonstrates that the `iterI >= 0` is not related to the lower_bound.

Reply:
how can it be the lower bound if the lower bound is a function argument?

Reply:
%lb and %step were removed from the func args.

Reply:
my question is how does changing %ld to be a constant helps demonstrate that iterI >= 0 predicate is not related to the lower_bound.

Reply:
I can revert that part (lb, step -> constants) but it should cover the dynamic loop with epilogue peeling either way.

Reply:
yeah I would remove it. I think it is good to test the general case where nothing is known about the loop bound